src/cache/cache_las.c

```diff
@@ -1397,6 +1397,15 @@ err:		__wt_buf_free(session, sweep_key);
 	return (ret);
 }
 
+/*
+ * When threads race modifying a record, we can end up with more than the max
+ * number of modifications in an update list.
+ *
+ * Leave space for double the maximum number on the stack as a best attempt to
+ * avoid heap allocation.
+ */
+#define	WT_MODIFY_ARRAY_SIZE	(WT_MAX_MODIFY_UPDATE * 2)
```

Comment:
My approach towards allocation is pretty much identical to https://github.com/wiredtiger/wiredtiger/blob/develop/src/btree/bt_ret.c#L137.
There might be some potential for code reuse there.

Reply:
We can explore code reuse once we start copying directly from the las cursor into the user cursor. I suggest leave a todo comment so that we do not forget.

Reply:
067612a

Added a todo.